.github/workflows/ci.yml

```diff
@@ -2,170 +2,77 @@ name: CI
 
 on:
   push:
-    branches: [ main ]
+    branches:
+      - main
   pull_request:
   merge_group:
-    types: [ checks_requested ]
 
 jobs:
-  ensure-no_std:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: nightly-2024-11-23
-          targets: wasm32-unknown-unknown
-      - uses: Swatinem/rust-cache@v2
-      - run: |
-          cd ensure-no_std && cargo build
-
-  parallel-tests:
-    runs-on: ubuntu-latest
+  build-test:
+    name: build test ${{ matrix.platform.name }}
+    runs-on: ${{ matrix.platform.os }}
     strategy:
+      fail-fast: false
       matrix:
-        cmd:
-          - test --profile=ci -p cairo-lang-casm --features=serde,parity-scale-codec,schemars
-          - test --profile=ci -p cairo-lang-casm --no-default-features --features=serde,parity-scale-codec
-          - test --profile=ci -p cairo-lang-compiler
-          - test --profile=ci -p cairo-lang-debug
-          - test --profile=ci -p cairo-lang-defs
-          - test --profile=ci -p cairo-lang-diagnostics
-          - test --profile=ci -p cairo-lang-doc
-          - test --profile=ci -p cairo-lang-eq-solver
-          - test --profile=ci -p cairo-lang-filesystem
-          - test --profile=ci -p cairo-lang-formatter
-          - test --profile=ci -p cairo-lang-language-server
-          - test --profile=ci -p cairo-lang-lowering
-          - test --profile=ci -p cairo-lang-parser
-          - test --profile=ci -p cairo-lang-plugins
-          - test --profile=ci -p cairo-lang-proc-macros
-          - test --profile=ci -p cairo-lang-project
-          - test --profile=ci -p cairo-lang-runnable
-          - test --profile=ci -p cairo-lang-runnable-utils
-          - test --profile=ci -p cairo-lang-runner
-          - test --profile=ci -p cairo-lang-semantic
-          - test --profile=ci -p cairo-lang-sierra
-          - test --profile=ci -p cairo-lang-sierra-ap-change
-          - test --profile=ci -p cairo-lang-sierra-gas
-          - test --profile=ci -p cairo-lang-sierra-generator
-          - test --profile=ci -p cairo-lang-sierra-to-casm
-          - test --profile=ci -p cairo-lang-sierra-type-size
-          - test --profile=ci -p cairo-lang-starknet-classes
-          - test --profile=ci -p cairo-lang-starknet
-          - test --profile=ci -p cairo-lang-syntax
-          - test --profile=ci -p cairo-lang-syntax-codegen
-          - test --profile=ci -p cairo-lang-test-plugin
-          - test --profile=ci -p cairo-lang-test-runner
-          - test --profile=ci -p cairo-lang-test-utils
-          - test --profile=ci -p cairo-lang-utils --features=serde,parity-scale-codec,schemars,testing,env_logger
-          - test --profile=ci -p cairo-lang-utils --no-default-features --features=serde,parity-scale-codec
-          - test --profile=ci -p tests
+        platform:
+          - name: linux x86-64
+            os: ubuntu-latest
```

Comment:
Does it need to be in the matrix though?

Reply:
I want to run test on Windows in the future

Reply:
premature optimization but i can live with that